core/src/main/scala/fs2/util/Task.scala

```diff
@@ -259,13 +259,20 @@ object Task extends Instances {
     ref[A].flatMap { ref => ref.set(t) map (_ => ref.get) }
 
   /**
+   Like [[async]], but run the callback in the same thread in the same
+   thread, rather than evaluating the callback using a `Strategy`.
+   */
+  def unforkedAsync[A](register: (Either[Throwable,A] => Unit) => Unit): Task[A] =
+    async(register)(Strategy.sequential)
+
+  /**
    Create a `Task` from an asynchronous computation, which takes the form
    of a function with which we can register a callback. This can be used
    to translate from a callback-based API to a straightforward monadic
-   version.
+   version. The callback is run using the strategy `S`.
    */
-  def async[A](register: (Either[Throwable,A] => Unit) => Unit): Task[A] =
-    new Task(Future.async(register))
+  def async[A](register: (Either[Throwable,A] => Unit) => Unit)(implicit S: Strategy): Task[A] =
```

Comment:
@pchiusano I think we will have to little bit streamline here. with recent PR by @mpilquist I wonder what you really do there is sort of `runFullyAsync`. I think it will be really mess for users what is `unforkedAsync`, `async` and `runFullyAsync` and their differences. Not sure if I undestand whats the difference between `runFullyAsync` and `async` ?